src/equations/compressible_euler_2d.jl

```diff
@@ -553,29 +553,111 @@ end
+  # Eleuterio F. Toro (2009)
+  # Riemann Solvers and Numerical Methods for Fluid Dynamics: A Pratical Introduction
+  # [DOI: 10.1007/b79761](https://doi.org/10.1007/b79761)
+  if v_normal <= 0.0
+    sound_speed = sqrt(equations.gamma * p_local / rho_local) # local sound speed
+    p_star = p_local * (1.0 + 0.5 * (equations.gamma - 1) * v_normal / sound_speed)^(2.0 * equations.gamma * equations.inv_gamma_minus_one)
+  else # v_normal > 0.0
+    A = 2.0 / ((equations.gamma + 1) * rho_local)
+    B = p_local * (equations.gamma - 1) / (equations.gamma + 1)
+    p_star = p_local + 0.5 * v_normal / A * (v_normal + sqrt(v_normal^2 + 4.0 * A * (p_local + B)))
+  end
+
+  # For the slip wall we directly set the flux as the normal velocity is zero
+  return SVector(zero(eltype(u_inner)),
+                 p_star * normal[1],
+                 p_star * normal[2],
+                 zero(eltype(u_inner))) * norm_
+end
+
+"""
+    boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
+                                 surface_flux_function, equations::CompressibleEulerEquations2D)
+Should be used together with [`TreeMesh`](@ref).
 
 !!! warning "Experimental code"
     This wall function can change any time.
 """
-@inline function boundary_state_slip_wall(u_internal, normal_direction::AbstractVector,
-                                          equations::CompressibleEulerEquations2D)
+function boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
+                                      surface_flux_function, equations::CompressibleEulerEquations2D)
+  # get the appropriate normal vector from the orientation
+  if orientation == 1
+    normal = SVector(1, 0)
+  else # orientation == 2
+    normal = SVector(0, 1)
+  end
+  # compute and return the flux using `boundary_condition_slip_wall` routine above
+  return boundary_condition_slip_wall(u_inner, normal, x, t, surface_flux_function, equations)
+end
 
-  # normalize the outward pointing direction
-  normal = normal_direction / norm(normal_direction)
+"""
+    boundary_condition_slip_wall(u_inner, normal_direction, direction, x, t,
+                                 surface_flux_function, equations::CompressibleEulerEquations2D)
+Should be used together with [`StructuredMesh`](@ref).
 
-  # compute the normal and tangential components of the velocity
-  u_normal  = normal[1] * u_internal[2] + normal[2] * u_internal[3]
-  u_tangent = (u_internal[2] - u_normal * normal[1], u_internal[3] - u_normal * normal[2])
+!!! warning "Experimental code"
+    This wall function can change any time.
+"""
+function boundary_condition_slip_wall(u_inner, normal_direction::AbstractVector, direction, x, t,
+                                      surface_flux_function, equations::CompressibleEulerEquations2D)
+
+  if iseven(direction) # normal is already outward pointing
+    normal = normal_direction
+  else # flip sign to make it outward pointing
+    normal = -normal_direction
+  end
```

Comment:
```suggestion
  if isodd(direction) # flip sign to make it outward pointing
    normal = -normal_direction
  else # normal is already outward pointing
    normal = normal_direction
  end
```
I would flip this around such that the case that requires action is first; this makes it IMHO easier to read since it is symmetric to the `if` condition below, where the sign of the normal flux needs to be flipped again.

Reply:
Alternatively, you could consider reducing this to a single conditional, which might be even better in terms of legibility
```julia
  # flip sign of normal to make it outward pointing, then flip the sign of the normal flux back to be inward pointing on
  # the -x and -y sides due to the orientation convention used by StructuredMesh
  if isodd(direction)
    boundary_flux = -boundary_condition_slip_wall(u_inner, -normal, x, t, surface_flux_function, equations)
  else
    boundary_flux = boundary_condition_slip_wall(u_inner, normal, x, t, surface_flux_function, equations)
  end

Reply:
I am open to this strategy. It also makes it such that we only use one if statement which should be faster too

Reply:
This worked and I think is cleaner so it is now in the current version